src/test/java/pages/MainPage.java

```diff
@@ -0,0 +1,29 @@
+package pages;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class MainPage extends BasePage {
+
+    @FindBy(className = "auth-bar__item--text")
+    public static WebElement entrance_Button;
+
+    @FindBy(xpath = "//div[@class='auth-form__label-part auth-form__label-part_1']/following::div[@class='auth-form__label-title']")
+    public static WebElement login_Via_NickName_Message;
+
+    public MainPage() {
+        super();
+    }
+
+    public void openPage() {
```

Comment:
Would it make sense to make BasePage abstract and define there openPage() method which will be overriden in the POs?

Reply:
no sense of making it abstract, fixed

Reply:
why no sense?

Reply:
my mistake. we can make it abstract and use loadable component and abstract method together to check if the page is opened and inherit this method in child classes